tests/platform_tests/api/test_psu.py

```diff
@@ -200,3 +200,33 @@ def test_led(self, duthost, localhost, platform_api_conn):
                         self.expect(color == color_actual, "Status LED color incorrect (expected: {}, actual: {}) from PSU {}".format(color, color_actual, psu_id))
         self.assert_expectations()
 
+    def test_get_position_in_parent(self, platform_api_conn):
+        for psu_id in range(self.num_psus):
+            position = psu.get_position_in_parent(platform_api_conn, psu_id)
+            self.expect(position is not None, "Failed to perform get_position_in_parent for psu id {}".format(psu_id))
+            self.expect(isinstance(position, int), "Position value must be an integer value for psu id {}".format(psu_id))
```

Comment:
The second `self.expect()` call is irrelevant if the first fails, so it should be nested using `if self.expect()` on the first. See lines 194-195 as an example.